service/controller/resource/promxy/servergroup/create.go

```diff
@@ -0,0 +1,61 @@
+package servergroup
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/promxy"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cluster, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configMapName := key.PromxyConfigMapName()
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking if %s configmap already exists ", configMapName))
+	configmap, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.PromxyConfigMapNamespace()).Get(ctx, configMapName, metav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("configmap %s does not exists", configMapName))
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	promxyConfiguration, err := promxy.Deserialize(configmap.Data["values"])
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	url, err := url.Parse("https://kubernetes.default:443")
```

Comment:
Pardon my ignorance. Does this work? What does it mean?

Reply:
Yes, basically, Kube creates a kubernetes service in the defautl namespace to access the api server. I am sure we can do better here but I wanted to make it work on friday. I had to add the api server url as https://github.com/prometheus/prometheus/blob/8dbfa14607a5cb138df43638f06c1a938c73df84/discovery/kubernetes/kubernetes.go#L167 requires the API server to be set

Reply:
Cool, TIL 👆

Reply:
I changed it to use the host in the k8s client but now you know :D

Reply:
If we can rely on kube doing this, I'd say this is actually a pretty nice and consistent way to access the API server. But sure, up to you.